doc/secure_boot/Secureboot_uefi_key_management.md

```diff
@@ -0,0 +1,217 @@
+
+#### Authenticated variable creation
+```
+Sample example of authenticated variable creation:
+
+uuidgen --random > GUID.txt
+
+**PK authenticated variable:**
+openssl x509 -outform DER -in PK.crt -out PK.cer
+cert-to-efi-sig-list -g "$(< GUID.txt)" PK.crt PK.esl
+sign-efi-sig-list -g "$(< GUID.txt)" -k PK.key -c PK.crt PK PK.esl PK.auth
+
+**KEK authenticated variable:**
+openssl x509 -outform DER -in KEK.crt -out KEK.cer
+cert-to-efi-sig-list -g "$(< GUID.txt)" KEK.crt KEK.esl
+sign-efi-sig-list -g "$(< GUID.txt)" -k PK.key -c PK.crt KEK KEK.esl KEK.auth
+
+**Db authenticated variable:**
+openssl x509 -outform DER -in db.crt -out db.cer
+cert-to-efi-sig-list -g "$(< GUID.txt)" db.crt db.esl
+sign-efi-sig-list -g "$(< GUID.txt)" -k KEK.key -c KEK.crt db db.esl db.auth
+```
+
+### 4.2 Enrollment of authenticated variables into the device
+The enrollment interface depends on where the keys are stored. The UEFI keys enrollment follows a specific policy. UEFI keys are PK, KEK, db and dbX. The enrollment of PK depends on access over UEFI FW's mode(e.g.: Setup mode, User mode, etc.). Then the owner of PK can enroll KEK since KEK enrollment requires authenticated variables signed by PK. 
+1. Copy the generated authenticated variable(Ex. KEK.auth, db.auth) to the device
+2. Use CLI to send the authenticated variable to UEFI firmware.
+3. UEFI firmware checks the trust source and integrity of authenticated variables and enrolls into UEFI database.
+4. The system requires a reboot to activate the keys for use.
+
+### 4.3 Remove keys from UEFI database
+For removal of keys requires the removal of an authenticated variable outside of the system. 
+
+````
+Sample example of authenticated variable creation for removal of the key from the signature database list:
+E.g The UEFI database has 3 entries of keys and one of them to be removed here
+
+1. Read the db key list from UEFI DB. 
+   efi-readvar -v Db -o DB-orig.esl 
+2. Convert signature list to individual certs. This command extracts DB-orig-1.der DB-orig-2.der DB-orig-3.der so on key files
+   sig-list-to-certs DB-orig.esl DB-orig
+3. Convert each of the certs into pem format
+   openssl x509 -inform der -in db-orig-1.der -out db-orig-1.pem
+   openssl x509 -inform der -in db-orig-2.der -out db-orig-2.pem
+   openssl x509 -inform der -in db-orig-3.der -out db-orig-3.pem
+4. Removal of db-orig-2.pem to be discarded
+5. Convert db-orig-1.pem and db-orig-3.pem to esl format
+   cert-to-efi-sig-list -g <GUID> db-orig-1.pem db-orig-1.esl
+   cert-to-efi-sig-list -g <GUID> db-orig-3.pem db-orig-3.esl
+6. Concatenate these two esl files 
+   cat db-orig-1.esl db-orig-3.esl > DB-new.esl
+7. Create authenticated variable and sign it
+   sign-efi-sig-list -g "<GUID>" -k KEK.key -c KEK.crt db DB-new.esl DB-new.auth
+8. Copy DB-new.auth to the system and use CLI to update UEFI database
+
+````
+
+### 4.4 Revoke keys
```

Comment:
Worth mentioning that key revocation is a potentially destructive action.
If we are removing a DB entry that is currently being used - this will cause the system to enter fatal mode (recoverable only via UEFI).

Reply:
totally agree. See also my comment above about protection of boot chain.